### PR TITLE
Loading the Facility Page when a Facility Row is Clicked

### LINKED ISSUE #126: Load the Facility Form when a Facility Row is Clicked On
In order to allow users to edit facility data, we should load the Facility form for a given facility when they click on that facility.

Note:  Any edits made in the form should be saved and reflected on the dashboard page when the user clicks save

### PR DESCRIPTION
**Pull request by sprad**
Loads the Facility page when a Facility Row on the Multifacility page is clicked

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #126

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/entry-point-client/App.tsx [hunk 1]
```diff
@@ -1,37 +1,43 @@
+import React, { useState } from "react";
 import { hot } from "react-hot-loader";
 import { Route, Switch } from "react-router-dom";
 
 import AuthWall from "../auth/AuthWall";
 import { LocaleDataProvider } from "../locale-data-context";
+import { FacilityContext } from "../page-multi-facility/FacilityContext";
```
Comment (sprad) [inline src/entry-point-client/App.tsx:7]: Clearly this file shouldn't depend on something in the "page-multi-facility" package, but `FacilityContext` should also live in a more globally accessible package.

File: src/page-multi-facility/FacilityPage.tsx [hunk 2]
```diff
@@ -1,14 +1,16 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
```
Comment (sprad) [inline src/page-multi-facility/FacilityPage.tsx:1]: Some of these aren't used so I'll delete.  Not sure why the linter didn't pick these up

File: src/page-multi-facility/FacilityRow.tsx [hunk 3]
```diff
@@ -15,25 +18,34 @@ const groupStatus = {
...
-  }
+  const history = useHistory();
+  const { setFacility } = useContext(FacilityContext);
+
+  const { id, name, updatedAt } = facility;
 
-  const { name } = facility;
+  const openFacilityPage = (event: React.MouseEvent<HTMLElement>) => {
+    setFacility(facility);
+    history.push("/multi-facility/facility");
```
Comment (sprad) [inline src/page-multi-facility/FacilityRow.tsx:34]: This is the key code to getting this all to work.  When a facility row is clicked, we set that information in a globally accessible context.  Then when we load the multi-facility page that facility is available to be used in the form in an update fashion.  If a facility is not present on the facility page, then we can create a new facility in the empty form.

**Comment by sprad** [comment]
This isn't a huge PR, but as someone not super familiar with React it was... a journey to get here.

Is the new Facility page still a WIP?  It seems like the page was updated to prioritize the form vs. the chart instead of allowing them to split the page.  This seems off to me.

Screen Shot 2020-04-18 at 8 23 22 PM

File: src/page-multi-facility/FacilityContext.tsx [hunk 4]
```diff
@@ -2,6 +2,12 @@ import React from "react";
 
 import { Facility } from "./types";
 
-export const FacilityContext = React.createContext<Facility | undefined>(
-  undefined,
-);
+interface FacilityContextProps {
+  facility?: Facility;
+  setFacility?: any;
```
Comment (sprad) [inline src/page-multi-facility/FacilityContext.tsx:7]: Is there a "type" for a function?

**Comment by jessex** [comment]
@sprad yes that page still needs its styling finalized, and some of the form components need to be removed until a later release. So still a WIP.

File: src/page-multi-facility/FacilityContext.tsx (hunk 4 above)
Comment (jessex) [reply src/page-multi-facility/FacilityContext.tsx:7]: I think this should be `void`

**Comment by sprad** [comment]
I just added some code to remove the Add Facility modal and have replaced it with a button that takes a user to the Facility Page in the "create" (empty) state.

**Pull request merged**